File: app/src/main/java/com/example/anchornotes/data/db/TagDao.java
package com.example.anchornotes.data.db;

import androidx.room.Dao;
import androidx.room.Insert;
import androidx.room.OnConflictStrategy;
import androidx.room.Query;

import java.util.List;

@Dao
public interface TagDao {
    @Query("SELECT * FROM tags ORDER BY name ASC")
    List<TagEntity> getAll();

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    long insert(TagEntity tag);

    @Query("SELECT * FROM tags WHERE name = :name LIMIT 1")
    TagEntity getByName(String name);
}

File: app/src/main/java/com/example/anchornotes/data/db/TagDao.kt
package com.example.anchornotes.data.db

import androidx.lifecycle.LiveData
import androidx.room.Dao
import androidx.room.Insert
import androidx.room.OnConflictStrategy
import androidx.room.Query

/**
 * DAO for tag operations.
 */
@Dao
interface TagDao {
    @Query("SELECT * FROM tags ORDER BY name ASC")
    fun getAllTags(): LiveData<List<TagEntity>>
    
    @Insert(onConflict = OnConflictStrategy.REPLACE)
    fun insert(tag: TagEntity): Long
}

/**
 * DAO for note-tag cross-reference operations.
 */
@Dao
interface NoteTagCrossRefDao {
    @Insert(onConflict = OnConflictStrategy.REPLACE)
    fun insert(crossRef: NoteTagCrossRef)
}

